libraries/program-error/derive/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+//! Crate defining a procedural macro for building Solana program errors
+
+#![allow(clippy::integer_arithmetic)]
```

Comment:
nit: Is this `allow` required? If yes, could you add a comment explaining why?

Reply:
Nope apparently not, not sure why I even had it in there in the first place....

Reply:
It's required in the derive sub-crate, that was why !